include/SFML/Window/Event.inl

```diff
@@ -0,0 +1,67 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2023 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+Event::Event(const T& t)
+{
+    if constexpr (isEventType<T>)
+        m_data = t;
+    else
+        static_assert(isEventType<T>, "T must be a subtype of sf::Event");
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+bool Event::is() const
+{
+    if constexpr (isEventType<T>)
+        return std::holds_alternative<T>(m_data);
+    else
+        static_assert(isEventType<T>, "T must be a subtype of sf::Event");
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+const T& Event::get() const
+{
+    if constexpr (isEventType<T>)
+        return std::get<T>(m_data);
+    else
+        static_assert(isEventType<T>, "T must be a subtype of sf::Event");
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+const T* Event::getIf() const
+{
+    if constexpr (isEventType<T>)
+        return std::get_if<T>(&m_data);
+    else
```

Comment:
It's there for better error messages. If you remove that `else` then you get dramatically longer compiler errors that make it hard to find the failed `static_assert`. We trade some extra library complexity for better user ergonomics.

Reply:
I think eXpl0it3r suggests to remove the else but keep the static assert and that would work the same.
Maybe it would be more readable this way:
```c++
static_assert(isEventType<T>, "T must be a subtype of sf::Event");
if constexpr (isEventType<T>)
    return std::get_if<T>(&m_data);
```

Reply:
Oh okay. I just tried that and yes that works! Great idea.

Reply:
Is the `if constexpr` even needed if the condition has been checked one line before?

Reply:
It’s not necessary but it produces better error messages when the template parameter is invalid.